cmd/log-collector/cmd/cmd_test.go

```diff
@@ -28,6 +28,7 @@ var _ = Describe("log collector cmd helper unit tests", func() {
 			err := command.PersistentPreRunE(command, []string{})
 			Expect(err).Should(BeNil())
 			Expect(logCollector.KubeConfig).ShouldNot(BeEmpty())
+			Expect(logCollector.KubeConfig).Should(Equal(os.Getenv(internal.KubeconfigEnv)))
```

Comment:
Do not assert the field value with the os env var's value. We want to check if the env var's value is set in the struct field.
Set the value of env var before initializing the command on line 27. Then check whether the field value is initialized with correct value. You can store this value in some variable like `testKubeConfigPath`